Orange/widgets/utils/itemmodels.py

```diff
@@ -646,6 +647,60 @@ def python_variable_tooltip(self, var):
         text += self.variable_labels_tooltip(var)
         return text
 
+
+class DomainModel(VariableListModel):
+    ATTRIBUTES, CLASSES, METAS = 1, 2, 4
+    MIXED = ATTRIBUTES | CLASSES | METAS
+    SEPARATED = (CLASSES, PyListModel.Separator,
+                 METAS, PyListModel.Separator,
+                 ATTRIBUTES)
+    PRIMITIVE = (DiscreteVariable, ContinuousVariable)
+
+    def __init__(self, order=SEPARATED, valid_types=None, alphabetical=False):
+        super().__init__()
+        if isinstance(order, int):
+            order = [order]
+        self.order = order
+        self.valid_types = valid_types
+        self.alphabetical = alphabetical
+        self.set_domain(None)
+
+    def set_domain(self, domain):
+        self.beginResetModel()
+        content = []
+        # The logic related to separators is a bit complicated: it ensures that
+        # even when a section is empty we don't have two separators in a row
+        # or a separator at the end
+        add_separator = False
+        for section in self.order:
+            if section is self.Separator:
```

Comment:
I did it that way in the scatter plot because the rest is also separated. If I had just `"(None)", ATTRIBUTES)`, I wouldn't separate it.

It's not about space, it's about aesthetics. And freedom. :)